From 288dd2ae3495c7b344f966f88f0adf0ea8a7c512 Mon Sep 17 00:00:00 2001
From: Todd Whiteman <todd.whiteman@joyent.com>
Date: Fri, 27 Jul 2018 13:44:04 -0700
Subject: [PATCH] TRITON-642 cnapi crashing every hour - "Cannot read property
 'warn' of undefined" Reviewed by: Orlando Vazquez <orlando@joyent.com>
 Approved by: Orlando Vazquez <orlando@joyent.com>

---
 lib/models/waitlist.js | 7 +++----
 package.json           | 2 +-
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/lib/models/waitlist.js b/lib/models/waitlist.js
index 8d5d454..047d80f 100644
--- a/lib/models/waitlist.js
+++ b/lib/models/waitlist.js
@@ -99,7 +99,7 @@ function WaitlistDirectorStart() {
     }
 
     function cleanupIntervalFn() {
-        WaitlistDirector.cleanupOldTickets(onTicketsCleanedUp);
+        WaitlistDirector.cleanupOldTickets(self.log, onTicketsCleanedUp);
     }
 
     // This gets called every time we check and find tickets that have been
@@ -405,13 +405,12 @@ function ModelWaitlistTicketsUpdatedSince(opts, callback) {
  * Periodically purge old finished/expired waitlist tickets from moray.
  */
 
-WaitlistDirector.cleanupOldTickets = function (callback) {
-    var self = this;
+WaitlistDirector.cleanupOldTickets = function (log, callback) {
     var ts = Date.now();
     var then = ts - WAITLIST_CLEANUP_MAX_AGE_MS;
     var thenDate = new Date(then);
     var escts = common.filterEscape(thenDate.toISOString());
-    self.log.warn('cleaning up tickets with updated_at older than = %s', escts);
+    log.warn('cleaning up tickets with updated_at older than = %s', escts);
 
     var filter = sprintf(
         '(&' +
diff --git a/package.json b/package.json
index 7b04baa..0411506 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cnapi",
   "description": "SmartDataCenter Compute Node API",
-  "version": "1.11.0",
+  "version": "1.11.1",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
-- 
2.21.0

